### PR TITLE
chore: ci setup

### LINKED ISSUE #7: ci setup

### PR DESCRIPTION
**Pull request by talwinder50**
closes #7

Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>

### CODE REVIEW

File: ci/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+jobs:
+  - job: UnitTest
+    pool:
+      vmImage: ubuntu-16.04
+    timeoutInMinutes: 30
+    steps:
+      - template: azp-dependencies.yml
+      - checkout: self
+      - script: make checks && make unit-test
```
Comment (troyronda) [inline ci/azure-pipelines.yml]: This won't work - you don't have a unit-test target in the Makefile yet.

File: ci/azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        displayName: Run checks and unit test
+      - script: bash <(curl https://codecov.io/bash)
```
Comment (troyronda) [inline ci/azure-pipelines.yml]: You don't have a code coverage output from unit-test yet.

File: ci/azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+name: $(SourceBranchName)-$(Date:yyyyMMdd).$(Rev:r)
+variables:
+  GO_VERSION: 1.13
+  GOPATH: $(Agent.BuildDirectory)/go
+
+jobs:
+  - job: UnitTest
+    pool:
+      vmImage: ubuntu-16.04
```
Comment (troyronda) [inline ci/azure-pipelines.yml]: Use` ubuntu-18.04`

Comment (talwinder50) [reply ci/azure-pipelines.yml]: done

File: ci/azure-pipelines.yml (hunk 1 above)
Comment (talwinder50) [reply ci/azure-pipelines.yml]: added

Comment (fqutishat) [reply ci/azure-pipelines.yml]: you should spilt it to two target checks and uint-test

File: ci/azp-dependencies.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+
+steps:
+  - task: GoTool@0
+    displayName: 'Use Go $(GO_VERSION)'
+    inputs:
+      version: $(GO_VERSION)
+      goPath: $(GOPATH)
+  - script: echo "##vso[task.setvariable variable=PATH]$(Agent.BuildDirectory)/go/bin:${PATH}"
```
Comment (fqutishat) [inline ci/azp-dependencies.yml]: what is that

Comment (troyronda) [reply ci/azp-dependencies.yml]: This adds ~/go/bin to the PATH.

**Pull request merged**